syntax = "proto3";

package Imagine_Rpc.Internal;

message InternalMessage {
    enum Identity{
        Client = 0;
        Server = 1;
        ZooKeeper = 2;
    }
    enum Request{
        Register = 0;
        DeRegister = 1;
        Heartbeat = 2;
        Search = 3;
        Response = 4;
    }
    Identity send_identity_ = 1;                // 发送者身份
    Identity recv_identity_ = 2;                // 接收者身份
    string service_name_ = 3;                   // register/deregister: 注册的服务名|____________|search: 查找的服务名
    string method_name_ = 4;                    // register/deregister: 注册的方法名|____________|search: 查找的方法名
    Request request_type_ = 6;                  // 请求类型
    string server_ip_ = 9;                      // server的ip
    string server_port_ = 10;                   // server的port
}